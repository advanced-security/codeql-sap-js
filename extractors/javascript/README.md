# Extension to the JavaScript extractor to support CDS compilation

This directory contains a `pre-finalize.sh` script that can be dropped into the `tools` directory
of a JavaScript extractor to enable the automatic compilation and extraction of SAP CAP CDS files
during the `codeql database create` process.

The script requires that the `cds` extractor is available when `codeql database init` is called. This can either be through providing the extractors parent directory using the `--search-path` bundle, or by including the extractor (and this pre-finalize script) in a custom bundle created by the [CodeQL Development Toolkit (qlt)](https://github.com/advanced-security/codeql-development-toolkit).