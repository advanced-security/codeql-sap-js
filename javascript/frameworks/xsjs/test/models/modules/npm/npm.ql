import javascript

from CryptographicKeyCreation crypto
select crypto
