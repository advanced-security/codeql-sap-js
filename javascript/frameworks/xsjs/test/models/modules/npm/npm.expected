| npm.xsjs:3:14:3:71 | crypto. ... 1024 }) |
| npm.xsjs:5:15:5:72 | crypto. ... 4096 }) |